### PR TITLE
feat: generate provenance for legitify release

### PR DESCRIPTION
**Pull request by gal-legit**
#### What's being changed?
Generate SLSA provenance for releases.

#### Is this PR related to an existing issue?
#36 

#### Check off the following:

- [X] This PR follows the CONTRIBUTION.md guidelines
- [X] I have self-reviewed my changes before submitting the PR

### CODE REVIEW

File: .goreleaser.yaml [hunk 1]
```diff
@@ -52,7 +52,7 @@ archives:
 
 checksum:
   name_template: '{{ .ProjectName }}_{{ .Version }}_SHA512SUMS'
```
Comment (royb-legit) [inline .goreleaser.yaml:54]:
```suggestion
  name_template: '{{ .ProjectName }}_{{ .Version }}_SHA256SUMS'
```

**Review by royb-legit — COMMENTED** [review]
Add short description to readme

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -25,9 +27,30 @@ jobs:
...
+          hashes=$(echo $ARTIFACTS | jq --raw-output '.[] | {name, "digest": (.extra.Digest // .extra.Checksum)} | select(.digest) | {digest} + {name} | join("  ") | sub("^sha256:";"")' | base64 -w0)
+          echo "hashes=$hashes" >> $GITHUB_OUTPUT
+
+  provenance:
+    needs: [goreleaser]
+    permissions:
+      actions: read # To read the workflow path.
+      id-token: write # To sign the provenance.
+      contents: write # To add assets to a release.
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.2.2
```
Comment (royb-legit) [inline .github/workflows/release.yaml:53]: use hash

Comment (gal-legit) [reply .github/workflows/release.yaml:53]: unfortunately, it's not supported currently. See:
https://github.com/slsa-framework/slsa-github-generator/blob/main/README.md#referencing-slsa-builders-and-generators

File: README.md [hunk 3]
```diff
@@ -25,6 +25,16 @@ https://user-images.githubusercontent.com/74864790/178964716-825840a6-d714-4b1d-
 git clone git@github.com:Legit-Labs/legitify.git
 go run main.go analyze ...
 ```
+
+## Provenance
+As of v0.1.6, every legitify release contains a [SLSA Level 3 Provenacne](https://github.com/slsa-framework/slsa-github-generator/blob/main/internal/builders/generic/README.md) document.
```
Comment (royb-legit) [inline README.md:30]:
```suggestion
To enhance the software supply chain security of legitify's users, as of v0.1.6, every legitify release contains a [SLSA Level 3 Provenacne](https://github.com/slsa-framework/slsa-github-generator/blob/main/internal/builders/generic/README.md) document.  
```

**Review by royb-legit — APPROVED** [review]

**Review by royb-legit — APPROVED** [review]

**Pull request merged**